lightning/src/ln/channelmanager.rs

```diff
@@ -2196,7 +2196,7 @@ impl<ChanSigner: ChannelKeys, M: Deref> ChannelManager<ChanSigner, M> where M::T
 					return Err(MsgHandleErrInternal::send_err_msg_no_close("Got a message for a channel from the wrong node!", msg.channel_id));
 				}
 				if (msg.failure_code & 0x8000) == 0 {
-					try_chan_entry!(self, Err(ChannelError::Close("Got update_fail_malformed_htlc with BADONION not set")), channel_state, chan);
+					try_chan_entry!(self, Err(ChannelError::<ChanSigner>::Close("Got update_fail_malformed_htlc with BADONION not set")), channel_state, chan);
```

Comment:
1.22 is sad here (and the next hunk). I think you may have to just do eg
```
					let channel_error: ChannelError<ChanSigner> = ChannelError::Close("Got update_fail_malformed_htlc with BADONION not set");
					try_chan_entry!(self, Err(channel_error), channel_state, chan);
```

Reply:
Fixed.